src/software/ai/navigator/path_planner/hrvo/agent.h

```diff
@@ -0,0 +1,157 @@
+
+
+    /**
+     * Update the agent's path and current state
+     *
+     * @param time_step the current time step that the simulator is running at
+     */
+    void update(double time_step);
+
+
+    /**
+     * Update the agent's radius based on its current (robot_state) velocity
+     */
+    void updateRadiusFromVelocity();
+
+
+    /**
+     * Update the primitive which this agent is currently pursuing.
+     *
+     * @param new_primitive The new primitive to pursue
+     * @param world The world in which the new primitive is being pursued
+     * @param time_step the time_step to use to step at
+     */
+    virtual void updatePrimitive(const TbotsProto::Primitive &new_primitive,
+                                 const World &world, double time_step) = 0;
+
+    /**
+     * Computes the preferred velocity of this agent.
+     *
+     * @param time_step
+     * @return the computed preferred velocity
+     */
+    virtual Vector computePreferredVelocity(double time_step) = 0;
+
+    /**
+     * Run the HRVO algorithm, and compute the new velocity this agent should move at
+     *
+     * @param robots robots in the simulation
+     * @param time_step
+     */
+    virtual void computeNewVelocity(
+        std::map<unsigned int, std::shared_ptr<Agent>> &robots, double time_step) = 0;
+
+
+    /**
+     * Create the VO to represent the given agent, relative to this agent
+     *
+     * @param other_agent The Agent which this velocity obstacle is being generated for
+     * @return The velocity obstacle which other_agent should see for this Agent
+     */
+    virtual VelocityObstacle createVelocityObstacle(const Agent &other_agent) = 0;
+
+
+    /**
+     * Return the state of the agent
+     *
+     * @return The RobotState for this agent
+     */
+    RobotState getRobotState();
+
+    /**
+     * Get the preferred velocity for this robot
+     *
+     * @return the computed preferred velocity, if there is one.
+     */
+    Vector getPreferredVelocity() const;
+
+    /**
+     * Gets the max speed for this agent
+     *
+     * @return max speed for this agent
+     */
+    double getMaxSpeed() const;
+
+    /**
+     * Gets the radius for this agent
+     *
+     * @return radius for this agent
+     */
+    double getRadius() const;
+
+    /**
+     * Gets the the path for this agent
+     *
+     * @return Path for this agent
+     */
+    RobotPath &getPath();
+
+    /**
+     * Return the max acceleration of the agent
+     *
+     * @return The max acceleration of the agent
+     */
+    double getMaxAccel() const;
+
+    /**
+     * Set this robots preferred velocity
+     *
+     * @param The velocity for which preferred_velocity should be set to
+     */
+    void setPreferredVelocity(Vector velocity);
+
+    // robot id of this Agent
+    RobotId robot_id;
+    // current state
+    RobotState robot_state;
+    // whether this Agent is FRIENDLY or ENEMY
+    TeamSide side;
```

Comment:
Doesn't really seem like this is used. We can probably remove it (use in various classes + `hrvo_test.cpp` + javadocs)